NDVItimeseries.R

```diff
@@ -0,0 +1,527 @@
+library(tidyverse)
+library(haven) # for read_dta
+library(tidylog) # for output on operations/changes with select, drop, merges, etc.
+library(rio)
+library(janitor)
+library(sf) # for st_read
+
+library(openxlsx)
+library(readxl)
+library(moderndive) 
+library(readr)
+library(tidyr)
+library(lubridate) # for dates
+
+library(reshape2)
+library(ggplot2) # included in tidyverse
+library(ggthemes)
+library(stringr)
+library(dplyr)
+library(openxlsx)
+library(readr)
+library(sf)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Load Data -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+#admin 2
+
+NDVI1admin2 <-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/NDVIMean_ts_niger (1981-2000).csv")
+
+NDVI2admin2<-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/NDVIMean_ts_niger (2001-2022).csv")
+
+#admin 3
+
+NDVI1admin3 <-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/l3NDVIMean_ts_niger (1981-2000).csv")
+
+NDVI2admin3 <- read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/l3NDVIMean_ts_niger (2001-2022).csv")
+
+#merge datasets
+NDVIadmin2 <- dplyr::left_join(NDVI1admin2, NDVI2admin2)
+NDVIadmin3 <- dplyr::left_join(NDVI1admin3, NDVI2admin3)
+
+# geospatial
+nigerMapAdmin2 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 2 (1)/wb_niger_admin2_shapefile/niger_admin2.shp") 
+
+
+nigerMapAdmin3 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 3 (1)/NER_adm03_feb2018.shp")
+
+
+# Don't show scientific notation
+options(scipen = 999)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Clean titles and data -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("_NDVI", "", colnames(NDVIadmin2)[col])
+}
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("X", "", colnames(NDVIadmin2)[col])
+}
+
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("_NDVI", "", colnames(NDVIadmin3)[col])
+}
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("X", "", colnames(NDVIadmin3)[col])
+}
+head(NDVIadmin3)
+
+
+# Convert data from Wide to Long ---- 
+ndviData2 <- 
+  NDVIadmin2 %>%
+  select(admin1Name, admin1Pcod, 
+         admin2Name, admin2Pcod, "19810701": "20191231") 
+
+ndviDataLong2 <- 
+  gather(ndviData2, 
+         key = Date, 
+         value = ndvi, 
+         "19810701":"20191231") %>% 
+  mutate(Date = ymd(Date)) #convert from char to dates
+# "19810701":"20191231") 
+
+ndviDataLong2$ndvi <- as.numeric(ndviDataLong2$ndvi) 
```

Comment:
we are adjusting scale by 0.0001, so we can add a section for this

Reply:
You can combine these two together *and* use the tidyverse syntax as follows:

```ndviDataLong2 <- 
      ndviDataLong2  %>%
mutate(newndvi =  as.numeric(ndvi) *0.0001) ``

- Though I'd also recommend *not* using the same filename/overwriting the same file, and instead use interim variable names.
- tidyverse syntax/style guides recommends keeping variables all lower case (will also make your life easier): https://style.tidyverse.org/syntax.html
- here's also good place to add a sanity check/test to make sure the values are in the range that you're expecting them to be in.